src/main/scala/mesosphere/marathon/core/task/termination/impl/KillStreamWatcher.scala

```diff
@@ -22,10 +23,16 @@ object KillStreamWatcher extends StrictLogging {
     *
     * See [[mesosphere.marathon.stream.EnrichedSource$.eventBusSource]]
     */
-  private def killedInstanceIds(eventStream: akka.event.EventStream): Source[Instance.Id, Cancellable] = {
+  private def killedInstanceIds(eventStream: akka.event.EventStream, instancesToCheck: Iterable[Instance]): Source[Instance.Id, Cancellable] = {
     // MaxValue causes a dynamically sized buffer to be used.
     val bufferSize = Int.MaxValue
     val overflowStrategy = OverflowStrategy.fail
+
+    val alreadyConsideredTerminal = instancesToCheck
+      .filter(instance => considerTerminal(instance.state.condition))
```

Comment:
in KillService.killInstances we use "isTerminal" to filter. Shouldn't we filter by the same thing? I think that considerTerminal and isTerminal is not exactly the same set of conditions

Reply:
I mean we do this `instance.tasksMap.values.withFilter(!_.isTerminal)` so maybe we should also in KillServiceActor do `instance.filter(i => i.tasksMap.values.exists(!_.isTerminal))` and then use these instance ids when passing ids to KillStreamWatcher. 

Thanks to that we filter the same thing in the same place once instead of twice on two different places.

WDYT?

Reply:
If you look at `killedViaInstanceChanged` source, you'll see that it also uses `considerTerminal` method. I'd keep it to make everything consistent.